src/Adapter/Carrier/Repository/CarrierRepository.php

```diff
@@ -69,73 +79,132 @@ public function add(Carrier $carrier): CarrierId
             CannotAddCarrierException::class
         );
 
-        return new CarrierId($carrierId);
+        return new CarrierId((int) $carrierId);
     }
 
-    public function updateInNewVersion(CarrierId $carrierId, Carrier $carrier): Carrier
+    public function update(Carrier $carrier, ShopConstraint $shopConstraint, int $errorCode): void
```

Comment:
I think you can simply use `updateObjectModel` and no need for the shop constraint parameter either